components/director/pkg/apperrors/errors.go

```diff
@@ -34,10 +42,18 @@ type NotUnique interface {
 	IsNotUnique()
 }
 
-type notUniqueError struct{}
+type notUniqueError struct {
+	identifier string
+}
+
+func NewNotUniqueError(identifier string) *notUniqueError {
+	return &notUniqueError{
+		identifier: identifier,
+	}
+}
 
 func (e *notUniqueError) Error() string {
-	return "unique constraint violation"
+	return fmt.Sprintf("Object %s is not unique", e.identifier)
```

Comment:
If identifier is empty, there will be 2 spaces. I would add simple if to append `%s ` if identifier is defined